x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java

```diff
@@ -493,131 +549,139 @@ public void runAsDenied(Authentication authentication, String action, TransportM
+    private static void restOrigin(RestRequest request, StringMapMessage logEntry) {
+        assert LOCAL_ORIGIN_FIELD_VALUE.equals(logEntry.get(ORIGIN_TYPE_FIELD_NAME)); // this is the default
+        final InetSocketAddress socketAddress = request.getHttpChannel().getRemoteAddress();
+        if (socketAddress != null) {
+            final String formattedAddress = NetworkAddress.format(socketAddress.getAddress());
+            logEntry.with(ORIGIN_TYPE_FIELD_NAME, REST_ORIGIN_FIELD_VALUE).with(ORIGIN_ADDRESS_FIELD_NAME, formattedAddress);
         }
+        // fall through to local_node default
     }
 
-    private static Optional<String> restOriginTag(ThreadContext threadContext) {
-        final InetSocketAddress restAddress = RemoteHostHeader.restRemoteAddress(threadContext);
-        if (restAddress == null) {
-            return Optional.empty();
+    private static void restUri(RestRequest request, StringMapMessage logEntry) {
+        if (Strings.hasLength(request.path())) {
+            logEntry.with(URL_PATH_FIELD_NAME, request.path());
+        }
+        if (request.params().isEmpty() == false) {
+            // TODO this is probably wrong, params are decoded and they might contain the delimiter
+            logEntry.with(URL_QUERY_FIELD_NAME, Strings.collectionToDelimitedString(request.params().entrySet(), "&"));
         }
-        return Optional.of(new StringBuilder("origin_type=[rest], origin_address=[").append(NetworkAddress.format(restAddress.getAddress()))
-                .append("]")
-                .toString());
     }
 
-    private static Optional<String> transportOriginTag(TransportMessage message) {
-        final TransportAddress address = message.remoteAddress();
-        if (address == null) {
-            return Optional.empty();
+    private static void restOrTransportOrigin(TransportMessage message, ThreadContext threadContext, StringMapMessage logEntry) {
+        assert LOCAL_ORIGIN_FIELD_VALUE.equals(logEntry.get(ORIGIN_TYPE_FIELD_NAME)); // this is the default
+        final InetSocketAddress restAddress = RemoteHostHeader.restRemoteAddress(threadContext);
+        if (restAddress != null) {
+            logEntry.with(ORIGIN_TYPE_FIELD_NAME, REST_ORIGIN_FIELD_VALUE)
+                    .with(ORIGIN_ADDRESS_FIELD_NAME, NetworkAddress.format(restAddress.getAddress()));
+        } else {
+            final TransportAddress address = message.remoteAddress();
+            if (address != null) {
+                logEntry.with(ORIGIN_TYPE_FIELD_NAME, TRANSPORT_ORIGIN_FIELD_VALUE)
+                        .with(ORIGIN_ADDRESS_FIELD_NAME, NetworkAddress.format(address.address().getAddress()));
+            }
         }
-        return Optional.of(
-                new StringBuilder("origin_type=[transport], origin_address=[").append(NetworkAddress.format(address.address().getAddress()))
-                        .append("]")
-                        .toString());
+        // fall through to local_node default
```

Comment:
by default all events originate locally unless one of restOrigin or restOrTransportOrigin is called to populate the origin details.